src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -127,65 +141,150 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
+    if (parentId !== '') {
+      cursor.current = visibleNodes.findIndex((el) => el.id === parentId);
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  /**
+   * Finds the first node starting with the given character.
+   * Starts at the cursors current position, and wraps around to the beginning of the menu if a match isn't found.
+   * @param {*} char The character to search by
+   */
+  const findNodeMatching = (char) => {
+    let sortedNodes = visibleNodes.slice(cursor.current + 1, visibleNodes.length);
+    sortedNodes = sortedNodes.concat(visibleNodes.slice(0, cursor.current));
+    const match = sortedNodes.find((el) => el.id[0].toUpperCase() === char);
+    if (match) {
+      cursor.current = visibleNodes.findIndex((el) => el.id === match.id);
+      currentNodeId.current = match.id;
+      focusCurrentNode();
+    }
+  };
+
   const handleKeyDown = (event, item) => {
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      event.preventDefault();
-      handleOnClick(event, item);
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        event.preventDefault();
+        handleOnClick(event, item);
+        break;
+      case KeyCode.KEY_DOWN:
+        event.preventDefault();
+        handleDownArrow();
+        break;
+      case KeyCode.KEY_UP:
+        event.preventDefault();
+        handleUpArrow();
+        break;
+      case KeyCode.KEY_RIGHT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).ariaExpanded === 'true') {
+            handleDownArrow();
+          } else if (document.getElementById(currentNodeId.current).ariaHasPopup && (!document.getElementById(currentNodeId.current).ariaExpanded || document.getElementById(currentNodeId.current).ariaExpanded === 'false')) {
+            handleOnClick(event, item);
+          }
+        }
+        break;
+      case KeyCode.KEY_LEFT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).ariaExpanded === 'true') {
+            handleOnClick(event, item);
+          } else if (!document.getElementById(currentNodeId.current).ariaHasPopup || !document.getElementById(currentNodeId.current).ariaExpanded || document.getElementById(currentNodeId.current).ariaExpanded === 'false') {
+            findParentNode();
+          }
+        }
+        break;
+      case KeyCode.KEY_HOME:
+        event.preventDefault();
+        cursor.current = 0;
+        currentNodeId.current = visibleNodes[cursor.current].id;
+        break;
+      case KeyCode.KEY_END:
+        event.preventDefault();
+        cursor.current = visibleNodes.length - 1;
+        currentNodeId.current = visibleNodes[cursor.current].id;
+        break;
+      case KeyCode.KEY_TAB:
```

Comment:
The tab key should function using the browser default, and should not advanced the focus within the treeview, but instead skip to the next tabindex of 0 in the DOM. So this can be removed.

Reply:
Currently there's one wdio test that expects tab to shift focus to the next node in the tree. Should I update it to use the arrows instead? https://github.com/cerner/terra-dev-site/blob/07c56d421382c0c8758db5fb2f00743a5cdf4379/tests/wdio/secondaryNavigation-spec.js#L47-L51

Reply:
Yes we'll want to update the tests to match the newly expected behavior.

Reply:
Just pushed the change for tab and updated the wdio tests.